src/layout/layout.zig

```diff
@@ -201,33 +201,45 @@ pub const TupleData = struct {
     }
 };
 
-/// Size and alignment information
-pub const SizeAlign = packed struct(u36) {
-    size: u30, // u30 can represent sizes up to 1 GiB
-    alignment: std.mem.Alignment, // u6 bits needed on this platform
+/// Roc's version of alignment that is limited to a max alignment of 16B to save bits.
+pub const RocAlignment = enum(u3) {
+    @"1" = 0,
+    @"2" = 1,
+    @"4" = 2,
+    @"8" = 3,
+    @"16" = 4,
+    _,
+
+    pub fn toByteUnits(a: RocAlignment) usize {
+        return @as(usize, 1) << @intFromEnum(a);
+    }
 
-    // Note: on 32-bit targets, there's an extra bit of uninitialized memory here,
```

Comment:
Note: Zig does not allow for this extra bit of uninitialized memory. On 32bit systems, `zig build test` would fail. We weren't running that before, so it was not caught. It is now caught by `zig build -Dno-bin` which is run on every system.

Fixed by just making the `RocAlignment` type suggested by this comment.